### ISSUE TITLE
fluidcore engine converter: Not all samples are defined in samples.asm

### ISSUE THREAD

**Issue by agargara** [utz82/ZX-Spectrum-1-Bit-Routines#2]
When using the fluidcore converter, it seems that not all samples get defined in samples.asm. For example when running with the included demo track, music.asm includes `smp6` but samples.asm does not define this, resulting in the following error when trying to assemble with pasmo:

```
ERROR on line 85 of file music.asm
ERROR: Symbol 'smp6' is undefined
```

Furthermore, the samples which are missing seem to change each time I run the converter. So sometimes it might actually compile without errors, if you're lucky.

**Comment by utz82** [comment]
Does this happen with the pre-build xm2fluid.exe? I think I fixed this 
in trunk a while back, but I haven't updated the release package yet.

If you have the possibility to do so, a quick fix would be to build 
xm2fluid yourself from the current source. Otherwise I'll look into 
building a new release, but that might take a couple of days.

**Comment by utz82** [comment]
Ah well, here you go. If everything is working as intended, you can simply drop your wip music.xm into this.

[fluidcore-new.zip](https://github.com/utz82/ZX-Spectrum-1-Bit-Routines/files/14229741/fluidcore-new.zip)

**Comment by agargara** [comment]
Wow, that was a fast reply! Thanks so much, I should have thought to try the pre-build version. Unfortunately the .zip you sent seems to be built for 32-bit windows but I'm 64-bit? I will try to compile myself when I get some more time :D

**Comment by utz82** [comment]
My bad, it's probably because I forgot to link statically. Try this.
[xm2fluid.exe.zip](https://github.com/utz82/ZX-Spectrum-1-Bit-Routines/files/14230892/xm2fluid.exe.zip)

**Comment by agargara** [comment]
The bad news is that .exe also doesn't run unfortunately,
the good news is I was able to compile an .exe that _does_ work using Mingw!
~~But the second bad news is I am still getting the same sample missing errors.~~

EDIT:
Never mind, I think the xm2fluid.cpp in the folder you sent was the old version. I replaced it with the new version from this repo, compiled and now it is working perfectly. Thank you for all the support!

**Comment by agargara** [comment]
Oh, I think we can probably close this now and consider the situation resolved. Thank you for all your work, your tools are very cool and super useful for music making!!

**Comment by utz82** [comment]
Ah, I see what went wrong, now that I got some sleep - the zip/exe I 
uploaded doesn't actually contain the fixed code.

I guess you've figured it out yourself by now - just to make sure 
though: Use the code directly from the git repo, not the the one from 
the release or the one I uploaded here.

Thanks for your help, and success with the music making!

Edit: Aaand now I see that you just wrote the exact same thing above. Note to self: Don't rely on Github's issue notifications, as they don't actually forward edits.
